### DISCUSSION TITLE
The token obtained through account and password login does not require a Client secret for JWT verification

### DISCUSSION THREAD

**Discussion by bieshaoxiong** [keycloak/keycloak#27927] [Q&A] [answered]
![image](https://github.com/keycloak/keycloak/assets/64937534/6090be2c-c521-435c-99b7-0e0c75da3b4d)
![image](https://github.com/keycloak/keycloak/assets/64937534/4a6c7bc4-bfcf-4a4c-b7fc-e7005bf7fd3f)

I obtained the token by logging in with my account and password，
I am configuring JWT authentication on. net, and my IssuerSigningKey can be verified by filling in anything randomly. Why is this? Does this mean that as long as the token format generated by keycloak is the same, it can be passed? Or is it that my understanding of JWT is incorrect
![image](https://github.com/keycloak/keycloak/assets/64937534/4747d2b2-f274-43f2-998e-9543fc7253ee)
And I also didn't see where to configure the key, I was using the version of keycloak 19

**Comment by justin-tay** [comment, marked as answer]
This has more to do with the dotnet library you are using rather than Keycloak as that is the one responsible for validating the token.

Either the library is not validating the signature, or it is using Keycloak's OpenID Endpoint Configuration metadata to discover the `jwks_uri` to get the public keys that Keycloak used to sign the tokens.

In Keycloak keys are configured in `Realm settings > Keys`

The `jwks_uri` is typically at `http://localhost:8080/auth/realms/protocol/openid-connect/certs` for a relying party to get the public keys.
